nni/retiarii/strategy/_rl_impl.py

```diff
@@ -100,7 +102,8 @@ def step(self, action):
                 if self.cur_step < self.num_steps else self.action_dim
         }
         if self.cur_step == self.num_steps:
-            model = get_targeted_model(self.base_model, self.mutators, self.sample)
+            with _thread_lock:
+                model = get_targeted_model(self.base_model, self.mutators, self.sample)
```

Comment:
could explain more, why there is data race here?

Reply:
Why mutator is mutating, it will rely on some of its member attributes. This can cause trouble if you are calling multiple `mutator.mutate` at the same time.

Reply:
got it